src/NLog/Targets/FileTarget.cs

```diff
@@ -615,7 +586,7 @@ protected override void CloseTarget()
         {
             base.CloseTarget();
 
-            foreach (string fileName in new List<string>(this.initializedFiles.Keys))
+            foreach (string fileName in initializedFiles.GetItems())
```

Comment:
I reviewed and the simplified implementation of the  ```IEnumerable<String> GetItems()``` method was the reason of the built (dbf7381) failure. 

I attempted the following implementations. 
``` C#
return initializedFiles.Keys;
```

``` C#
return initializedFiles.Keys.AsEnumerable<String>();
```

``` C#
foreach (var file in initializedFiles)
{
	yield return file.Key;
}
```

``` C#
var items = from file in initializedFiles
			select file.Key;
return items; 
```

All  the implementations caused an "Collection was modified; enumeration operation may not execute" exception to be thrown and failure of DateArchive_UsesDateFromCurrentTimeSource FileTargetTests test.